packages/frontend/components/scaffold-eth/Contract/WriteOnlyFunctionForm.tsx

```diff
@@ -0,0 +1,107 @@
+import { FunctionFragment } from "ethers/lib/utils";
+import { useState } from "react";
+import { useContractWrite, usePrepareContractWrite } from "wagmi";
+import { tryToDisplay } from "./utilsDisplay";
+import InputUI from "./InputUI";
+import { getFunctionInputKey, getParsedEthersError } from "./utilsContract";
+import { TxValueInput } from "./utilsComponents";
+import { toast } from "~~/components/scaffold-eth";
+import reactHotToast from "react-hot-toast";
+
+// TODO set sensible initial state values to avoid error on first render, also put it in utilsContract
+const getInitialFormState = (functionFragment: FunctionFragment) => {
+  const initialForm: Record<string, any> = {};
+  functionFragment.inputs.forEach((input, inputIndex) => {
+    const key = getFunctionInputKey(functionFragment, input, inputIndex);
+    initialForm[key] = "";
+  });
+  return initialForm;
+};
+
+interface IFunctionForm {
+  functionFragment: FunctionFragment;
+  contractAddress: string;
+}
+
+export const WriteOnlyFunctionForm = ({ functionFragment, contractAddress }: IFunctionForm) => {
+  const [form, setForm] = useState<Record<string, any>>(() => getInitialFormState(functionFragment));
+  const [txValue, setTxValue] = useState("");
+
+  const keys = Object.keys(form);
+
+  // TODO handle gasPrice
+  const { config, error: inputError } = usePrepareContractWrite({
+    addressOrName: contractAddress,
+    functionName: functionFragment.name,
+    contractInterface: [functionFragment],
+    args: keys.map(key => form[key]),
+    overrides: {
+      value: txValue,
+    },
+  });
+
+  const {
+    data: result,
+    isLoading,
+    writeAsync,
+  } = useContractWrite({
+    ...config,
+  });
+
+  const handleWrite = async () => {
+    // TODO Show more descriptive error message
+    if (inputError) {
+      const message = getParsedEthersError(inputError);
+      toast.error(message);
+    }
+
+    if (writeAsync) {
+      try {
+        await reactHotToast.promise(
+          writeAsync(),
```

Comment:
used this as of now, maybe in the future wrap the `writeAsync` inside our own function, something similar to Transactor or more general which handles promise